plugins/polio/api/lqas_im/lqas_data_manager.py

```diff
@@ -0,0 +1,477 @@
+
+        # Require an account for the manager to work
+        if self.account is None:
+            raise ValueError("An Account is required. Please provide either account or account_id parameter.")
+
+    def parse_json_and_create_lqas_activities(self, lqas_json_data):
+        data_to_parse = lqas_json_data["stats"]
+        rounds, districts = self._prefetch_rounds_and_districts(lqas_json_data)
+        total_created = 0
+        total_failed = 0
+        total_parsed = 0
+        districts_not_found = []
+        rounds_not_found = []
+        for obr_name, campaign_data in data_to_parse.items():
+            for round_data in campaign_data.get("rounds", []) or []:
+                try:
+                    round_obj = rounds.get(number=round_data["number"], campaign__obr_name=obr_name)
+                    # Create lqas models for round here
+                    self._create_lqas_activity(round_obj, obr_name, round_data)
+                    self._create_lqas_no_marks(round_obj, obr_name, round_data)
+                    self._create_lqas_absences(round_obj, obr_name, round_data)
+                    # Create district data for current round
+                    districts_results = round_data.get("data", {}) or {}
+                    for district_name, district_data in districts_results.items():
+                        district_id = district_data["district"]
+                        try:
+                            district = districts.get(id=district_id)
+                            lqas_entry = self._create_lqas_entry(
+                                round_obj=round_obj, district=district, district_data=district_data
+                            )
+                            self._create_caregiver_stats(lqas_entry, district_data)
+                        except OrgUnit.DoesNotExist as e:
+                            self.logger.warning(f"No district found for {district_name}, id:{district_id} ")
+                            districts_not_found.append(district_id)
+                            continue
+                    total_created += 1
+                    total_parsed += 1
+                except Round.DoesNotExist as e:
+                    self.logger.warning(
+                        f"Could not add LQAS data for {obr_name} Round {round_data['number']}: Round not found"
+                    )
+                    self.logger.warning(e)
+                    rounds_not_found.append(f"{obr_name} Round {round_data['number']}")
+                    total_failed += 1
+                    total_parsed += 1
+                    continue
+                except IntegrityError as e:
+                    self.logger.warning(e)
+                    total_failed += 1
+                    total_parsed += 1
+                    continue
+                except ValidationError as e:
+                    self.logger.warning(e)
+                    total_failed += 1
+                    total_parsed += 1
+                    continue
```

Comment:
You can save some lines here (nitpick, only mentioned once):

```suggestion
                except (IntegrityError, ValidationError) as e:
                    self.logger.warning(e)
                    total_failed += 1
                    total_parsed += 1
                    continue
```